### DISCUSSION TITLE
Collector with same name must have same type and same label names

### DISCUSSION THREAD

**Discussion by tomkerkhove** [PrometheusClientNet/Prometheus.Client#149] [Q&A] [answered]
Is the "Collector with same name must have same type and same label names" limitation due to the library or one of Prometheus?

I'm looking at providing info metrics where the amount of labels is different but is not possible because of this, so curious to learn why.

**Comment by phnx47** [comment, marked as answer]
> Client libraries MUST NOT allow users to have different label names for the same metric for Gauge/Counter/Summary/Histogram or any other Collector offered by the library.

Docs: https://prometheus.io/docs/instrumenting/writing_clientlibs/#labels

**Reply by tomkerkhove** [reply]
That's what I've figured - Thanks for the URL!
